disttask/framework/dispatcher/dispatcher.go

```diff
@@ -373,9 +373,21 @@ func (d *dispatcher) processNormalFlow(gTask *proto.Task) (err error) {
 		logutil.BgLogger().Warn("gen gTask flow handle failed, this type handle doesn't register", zap.Int64("ID", gTask.ID), zap.String("type", gTask.Type))
 		return d.updateTask(gTask, proto.TaskStateReverted, nil, retrySQLTimes)
 	}
-	metas, err := handle.ProcessNormalFlow(d.ctx, d, gTask)
+	prevSubtasks, err := d.taskMgr.GetSucceedSubtasksByStep(gTask.ID, gTask.Step)
 	if err != nil {
-		logutil.BgLogger().Warn("gen dist-plan failed", zap.Error(err))
+		logutil.BgLogger().Warn("get previous succeed subtasks failed", zap.Int64("ID", gTask.ID), zap.String("type", gTask.Type), zap.Error(err))
+		return d.updateTask(gTask, proto.TaskStateReverted, nil, retrySQLTimes)
+	}
+	prevSubtaskMetas := make([][]byte, 0, len(prevSubtasks))
+	for _, subtask := range prevSubtasks {
+		prevSubtaskMetas = append(prevSubtaskMetas, subtask.Meta)
+	}
+	metas, retryable, err := handle.ProcessNormalFlow(d.ctx, d, gTask, prevSubtaskMetas)
```

Comment:
What does `prevSubtasks` do? I don't think it's in use at the moment. In addition, this quantity may be very large, is there any other way to deal with it? For example, put it in `gTask` to reshard the task.

Rest LGTM

Reply:
prevSubtaskMetas contains the result of previous step

Reply:
This is not a good idea.

If we need to record all the previous sub task, we may need to record the whole sub-task context because it could fail on any sub-task. The sub-tasks should not depends on each other, they should all depends on another thing, such context or Meta.

Refer to 'Dependence Inversion Principle'.